torch/csrc/autograd/functions/utils.h

```diff
@@ -54,6 +59,33 @@ inline bool compute_requires_grad(Args&&... args) {
   return ComputeRequiresGrad().apply(std::forward<Args>(args)...).out;
 }
 
+struct AssertNoInferenceTensor : IterArgs<AssertNoInferenceTensor> {
+  using IterArgs<AssertNoInferenceTensor>::operator();
+  void operator()(const at::Tensor& tensor) {
```

Comment:
This should also handle TensorList no?
Otherwise the checks for things like `cat` won't be correct.

Reply:
Yea I think so, for example we currently generate code like below which means `compute_requires_grad` handles TensorList. 
```
  Tensor _cat(c10::DispatchKeySet ks, TensorList tensors, int64_t dim) {
    auto tensors_ = unpack(tensors, "tensors", 0);
    auto _any_requires_grad = compute_requires_grad( tensors );
```

Reply:
Ho IterArgs handles List and std::vector unpacking by default. Nice!